xdis/code.py

```diff
@@ -637,3 +640,63 @@ def code_has_star_star_arg(code):
     """Return True iff
     The code object has a variable keyword parameter (**kwargs-like)."""
     return (code.co_flags & 8) != 0
+
+
+# From: https://stackoverflow.com/questions/36330860/pythonically-check-if-a-variable-name-is-valid
+def is_valid_variable_name(name):
+    """Returns True iff
+    the argument is a valid Python variable name"""
+    if not re.fullmatch('^[_a-zA-Z][_0-9a-zA-Z]*$', name):
+        return False
+    try:
+        ast.parse('{} = None'.format(name))
```

Comment:
Note that since at this stage we already know that the name matches `[_0-9a-zA-Z]+` so this can't be fooled by having code in the name.

Reply:
One thing to note. Since xdis works for 2.7-present. We shouldn't use .format here; the old (and IMHO just as good for this purpose) "%" works.

Dunno what I'm gong to do for the python 2.4 branch of this which goes back to Python 2.4, but I'll deal with that later.

Reply:
Agree, that's actually how I usually write, this was just due to copy-pasting. Will update.